Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -75,23 +78,25 @@ internal class ConnectionPool {
     internal static let PoolClosedError = MongoError.LogicError(message: "ConnectionPool was already closed")
 
     /// Initializes the pool using the provided `ConnectionString`.
-    internal init(from connString: ConnectionString) throws {
-        let pool: OpaquePointer = try connString.withMongocURI { uriPtr in
-            guard let pool = mongoc_client_pool_new(uriPtr) else {
-                throw MongoError.InternalError(message: "Failed to initialize libmongoc client pool")
-            }
-            return pool
-        }
+    internal init(from connString: ConnectionString, executor: OperationExecutor) throws {
```

Comment:
It feels kinda weird to pass the executor in like this, but I wasn't exactly sure how to do this otherwise while still keeping the the pool creation being blocking encapsulated within `ConnectionPool`.

Reply:
yeah, I was going to say make it an `Operation` but then it would need a `Connection` to be executed so of course that doesn't work. it's weird but I'm not sure what else to do really.